src/Storages/IStorage.h

```diff
@@ -185,6 +185,18 @@ class IStorage : public std::enable_shared_from_this<IStorage>, public TypePromo
     using ColumnSizeByName = std::unordered_map<std::string, ColumnSize>;
     virtual ColumnSizeByName getColumnSizes() const { return {}; }
 
+    virtual std::expected<ColumnSizeByName, Exception> tryGetColumnSizes() const
```

Comment:
```suggestion
    virtual std::expected<ColumnSizeByName, Exception> tryGetColumnSizes() const
```

All new `try*` methods should not be virtual